core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -1043,24 +1043,25 @@ static TEE_Result check_client(struct tee_ta_session *s, const TEE_Identity *id)
 /*-----------------------------------------------------------------------------
  * Close a Trusted Application and free available resources
  *---------------------------------------------------------------------------*/
-TEE_Result tee_ta_close_session(uint32_t id,
+TEE_Result tee_ta_close_session(struct tee_ta_session *csess,
 				struct tee_ta_session_head *open_sessions,
 				const TEE_Identity *clnt_id)
 {
 	struct tee_ta_session *sess;
 	struct tee_ta_ctx *ctx;
 
-	DMSG("tee_ta_close_session(%x)", (unsigned int)id);
+	DMSG("tee_ta_close_session(0x%" PRIxVA ")",  (vaddr_t)csess);
 
-	if (id == 0)
+	if (csess == 0)
```

Comment:
Don't compare pointers with `0`, in this case `if (!csess)` would be preferable.